src/native/corehost/fxr/command_line.cpp

```diff
@@ -164,6 +164,19 @@ namespace
                 trace::verbose(_X("Application '%s' is not a managed executable."), app_candidate.c_str());
                 if (!exec_mode)
                 {
+                    // Check if this is a non-managed file with directory separator that exists
+                    // This should show a specific error instead of routing to CLI
+                    bool has_dir_separator = strchr(app_candidate.c_str(), DIR_SEPARATOR) != nullptr;
```

Comment:
This can call `app_candidate.find(DIR_SEPARATOR)` instead. `pal::string_t` is defined to be `std::string` or `std::wstring` based on the platform, so `strchr` will not work on all platforms.